clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/secured/ConfigurationUtils.java

```diff
@@ -228,4 +233,21 @@ public <T> T get(String name) {
         return (T) configs.get(name);
     }
 
+    // make sure the url is in the "org.apache.kafka.sasl.oauthbearer.allowed.urls" system property
+    public void throwIfURLIsNotAllowed(String urlConfig) {
+        String allowedLoginModuleProp = System.getProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG);
+        if (allowedLoginModuleProp == null) {
+            // by default, we accept all URLs
+            return;
```

Comment:
By default, we accept all in v3.x.

Reply:
Hello @showuon @mimaison,

Quick question : Why not keep a similar behaviour in v4.x or have a solution not to impose this behaviour (like wildcard, flag) ?
Because of the sudden change between 3.x and 4.0, and the impossibility of authorizing all urls in 4.0

Thank you for your answer